argmin/src/core/state/iterstate.rs

```diff
@@ -35,7 +35,7 @@ use std::collections::HashMap;
 /// * termination status
 #[derive(Clone, Default, Debug, Eq, PartialEq)]
 #[cfg_attr(feature = "serde1", derive(Serialize, Deserialize))]
-pub struct IterState<P, G, J, H, F> {
+pub struct IterState<P, G, J, H, F, R> {
```

Comment:
I would prefer it if the precision `F` is the final parameter: `IterState<P, G, J, H, R, F>`